Python/Product/Analysis/Infrastructure/ProcessHelper.cs

```diff
@@ -99,34 +99,34 @@ sealed class ProcessHelper : IDisposable {
         }
 
         private void Process_ErrorDataReceived(object sender, DataReceivedEventArgs e) {
-            if (e.Data == null) {
-                try {
+            try {
+                if (e.Data == null) {
                     _seenNullError.Release();
-                } catch (ObjectDisposedException) {
+                } else {
+                    OnErrorLine?.Invoke(e.Data.TrimEnd());
                 }
+            } catch (ObjectDisposedException) {
                 ((Process)sender).ErrorDataReceived -= Process_ErrorDataReceived;
-                return;
             }
-
-            OnErrorLine?.Invoke(e.Data.TrimEnd());
         }
 
         private void Process_OutputDataReceived(object sender, DataReceivedEventArgs e) {
-            if (e.Data == null) {
-                try {
+            try {
+                if (e.Data == null) {
                     _seenNullOutput.Release();
-                } catch (ObjectDisposedException) {
+                } else {
+                    OnOutputLine?.Invoke(e.Data.TrimEnd());
                 }
+            } catch (ObjectDisposedException) {
                 ((Process)sender).OutputDataReceived -= Process_OutputDataReceived;
-                return;
             }
-
-            OnOutputLine?.Invoke(e.Data.TrimEnd());
         }
 
         public void Kill() {
             try {
-                _process?.Kill();
+                if (!_process.HasExited) {
```

Comment:
`_process?.Kill()` has null check. Do we need it here as well?

Reply:
@MikhailArkhipov Still want the null check here

Reply:
Done